### PR TITLE
Fix general regression issues and implement other requested changes 

### LINKED ISSUE #77: regression: Fix table formatting issue in `/mouse` and `/session` pages
# Mouse previous

image

# Mouse new

image

Primary issue is just the metadata row highlighting.

# Session previous

image

# Session new

image

Primary issue is:
- Table stretching near the bottom of the screenshot
- Metadata row highlighting

### LINKED ISSUE #81: Regression: add brain region filter that is text based (no tree)
Upgrade has caused filtering to move to backend so we need to move filtering to backend. It is currently being handled in the front end. For now, disabling the feature until we are ready.

### LINKED ISSUE #82: Regression: add public (brainwide, ephys, and behavior) and internal (brainwide, ephys, mouse delay, and behavior) check boxes back.
Upgrade has caused filtering to move to backend so we need to move filtering to backend. It is currently being handled in the front end. For now, disabling the feature until we are ready.

### LINKED ISSUE #96: Add Mouse status (alive/deceased) filter for sessions page
Good example is Mouse UUID `13fa4ee6-da26-4ecc-936a-2c3f49778f40`

### LINKED ISSUE #97: Add default sort state for `/sessions` table
Default should be `Ephys Availability` to be descending.

image

### LINKED ISSUE #98: Improve the experience for horizontal scrolling in `/sessions` table
A user reported they missed sorting by some attributes because it was not obvious that they could horizontally scroll. Although, if they scroll to the bottom they would have seen it. Perhaps the solution is to duplicate/move the horizontal scroll bar above the column headers.

## Top
image

# Bottom
image

### PR DESCRIPTION
**Pull request by zitrosolrac**
### Fix #77
**Before**
image
image
**After**
Screen Shot 2021-11-12 at 11 20 33 AM
Screen Shot 2021-11-12 at 10 44 33 AM
### Fix #98
**Before**
Screen Shot 2021-11-12 at 9 47 18 AM
**After**
Screen Shot 2021-11-12 at 9 46 00 AM
### Fix #97 
Screen Shot 2021-11-12 at 10 46 26 AM

### Fix #81 
Screen Shot 2021-11-12 at 10 46 59 AM

### Fix #82
Screen Shot 2021-11-12 at 10 47 16 AM
 
### Fix #96 
Screen Shot 2021-11-17 at 2 00 49 PM

### CODE REVIEW

File: ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts [hunk 1]
```diff
@@ -375,6 +391,10 @@ export class SessionListComponent implements OnInit, OnDestroy {
                 continue
               }
             }
+            if(key == 'brain_regions' && value !== null){
```
Comment (guzman-raphael) [inline ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts]: There appears to be a small bug related to 'unsetting' the brain filter after use. For instance, when I apply `HPF` as a filter, it appropriately restricts to 2 records but when I simply empty the text box, it shows 0 results.

File: ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts [hunk 2]
```diff
@@ -340,11 +341,26 @@ export class SessionListComponent implements OnInit, OnDestroy {
...
           if(this.sort.direction == ''){
-            this.sort.active = 'session_start_time';
+            this.sort.active = 'nprobe';
             this.sort.direction = 'desc'
           }
           let filter = Object.assign({}, this.session_filter_form.getRawValue());
-          let newFilter = JSON.stringify(filter)
+
+          if(this.hideMissingPlots){
+            filter["nplot"] = 1
```
Comment (guzman-raphael) [inline ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts]: Let's always make sure to terminate statements with a `;`. Please double-check your other changes here too.

```suggestion
            filter["nplot"] = 1;
```

File: ibl-frontend/frontend-content/src/app/session-list/session/session.component.css [hunk 3]
```diff
@@ -32,7 +32,7 @@ h5 {
 
 .fitParTable {
     font-size: 75%;
-    display: table-footer-group;
+    display: table-row;
```
Comment (guzman-raphael) [inline ibl-frontend/frontend-content/src/app/session-list/session/session.component.css:35]: Could you add a bit more details here on what this update was addressing? A comment here should suffice. Just trying to understand a bit how this impacted the fixes.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@zitrosolrac Awesome contributions! Really nice to see you getting the hang of this, Carlos. Excellent work. :muscle: 

Points raised here are quite minor so we should be ready for a merge soon. The main one is regarding a small bug in unsetting the `Brain Regions` filter.

We can likely merge it in our next tagup and start planning for the next deploy cycle.

File: ibl-frontend/frontend-content/src/app/session-list/session/session.component.css (hunk 3 above)
Comment (zitrosolrac) [reply ibl-frontend/frontend-content/src/app/session-list/session/session.component.css:35]: This had to deal with the formatting of the tables underneath the graphs of the session page. They can be seen on the bottom portion of this image. 
image

File: ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts (hunk 1 above)
Comment (zitrosolrac) [reply ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts]: This been address in the latest commit!

File: ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts (hunk 2 above)
Comment (zitrosolrac) [reply ibl-frontend/frontend-content/src/app/session-list/session-list.component.ts]: This has been resolved in the latest commit!

File: ibl-frontend/frontend-content/src/app/session-list/session/session.component.css (hunk 3 above)
Comment (guzman-raphael) [reply ibl-frontend/frontend-content/src/app/session-list/session/session.component.css:35]: Great, thanks. Makes sense.

Comment (zitrosolrac) [reply ibl-frontend/frontend-content/src/app/session-list/session/session.component.css:35]: You're welcome!

File: backend/iblapi.py [hunk 4]
```diff
@@ -250,6 +250,11 @@ def handle_q(subpath, args, proj, fetch_args=None, **kwargs):
             training_status='training_status', good_enough_for_brainwide_map='good_enough_for_brainwide_map',
             keep_all_rows=True
             )
+        subj = subject.Subject().aggr(
+            subject.Death().proj('death_date') * dj.U('death_date'),
+            # death_date='IF(death_date>0, 1, 0)',
```
Comment (guzman-raphael) [inline backend/iblapi.py]:
```suggestion
```

File: ibl-frontend/frontend-content/src/app/session-list/session-list.component.html [hunk 5]
```diff
@@ -155,11 +155,33 @@ <h4>
...
+          <!-- <mat-form-field>
+            <label>Alive/Deceased</label>
+            <input matInput type="text" name="death_date" formControlName="death_date"
+              [matAutocomplete]="autoCompRU" (focus)="stepBackMenu($event)">
+            <mat-autocomplete #autoCompRU="matAutocomplete">
+              <mat-option *ngFor="let death_date of dropDownMenuOptions.filteredResponsibleUserOptions | async" [value]="death_date"
+                (click)="updateMenu()">{{ death_date }}</mat-option>
+            </mat-autocomplete>
+          </mat-form-field> -->
+
```
Comment (guzman-raphael) [inline ibl-frontend/frontend-content/src/app/session-list/session-list.component.html]:
```suggestion
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@zitrosolrac This looks great man! :sunglasses: 

Just need to remove some unnecessary comments and we should be ready to merge. :+1:

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**